appinventor/blocklyeditor/src/blocks/control.js

```diff
@@ -166,6 +164,32 @@ Blockly.Blocks['controls_if'] = {
     }
   },
   typeblock: [{translatedName: Blockly.Msg.LANG_CONTROLS_IF_IF_TITLE_IF}],
+  rebuildShape_: function () {
+    console.log('rebuidShape');
+    var valueConnections = [null];
+    var statementConnections = [null];
+    var elseStatementConnection = null;
+
+    if (this.getInput('ELSE')) {
+      elseStatementConnection =
+          this.getInput('ELSE').connection.targetConnection;
+    }
+    for (var i = 1; this.getInput('IF' + i); i++) {
+      var inputIf = this.getInput('IF' + i);
+      var inputDo = this.getInput('DO' + i);
+      valueConnections.push(inputIf.connection.targetConnection);
+      statementConnections.push(inputDo.connection.targetConnection);
+    }
+    this.updateShape_();
+    this.reconnectChildBlocks_(valueConnections, statementConnections, elseStatementConnection);
+  },
+  reconnectChildBlocks_: function (valueConnections, statementConnections, elseStatementConnection) {
+    for (let i = 1; i <= this.elseifCount_; i++) {
+      valueConnections[i]?.reconnect(this, 'IF' + i);
+      statementConnections[i]?.reconnect(this, 'DO' + i);
+    }
+    elseStatementConnection?.reconnect(this, 'ELSE');
```

Comment:
Given how old of a browser some schools seem to get stuck with I'm a bit worried about using optional chaining as a language feature.

Reply:
If we do decide to go forward with this, we need to update the minimum browser version list in index.jsp to be:

```
Chrome 80+
Safari 13.1+
Firefox 74+
```

Reply:
> Given how old of a browser some schools seem to get stuck with I'm a bit worried about using optional chaining as a language feature.

Sorry about that.  I just copy-pasted from the lexvar code, where I was less concerned about using more modern language features.